### ISSUE TITLE
Webpack (and other executors) use the current target when generating a temporary tsconfig. 

### ISSUE THREAD

**Issue by statop** [nrwl/nx#21395]
### Current Behavior

If I use `@nx/webpack:webpack` in a target other than 'build' with `buildLibsFromSource: false` then it will use that other target when calculating the dependencies for generating the temporary tsonfig.

### Expected Behavior

If I use `@nx/webpack:webpack` in a target other than 'build' with `buildLibsFromSource: false` then it will use the 'build' target when calculating the dependencies for generating the temporary tsonfig.

### Steps to Reproduce

1.  use `@nx/webpack:webpack` in a target other than 'build' with `buildLibsFromSource: false` 

Note that the temporary tsconfig only contains mappings to the dist folder to dependencies that also have that target.

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node   : 18.17.1
   OS     : darwin-arm64
   pnpm   : 8.14.3

   nx                 : 17.2.8
   @nx/js             : 17.2.8
   @nx/jest           : 17.2.8
   @nx/eslint         : 17.2.8
   @nx/workspace      : 17.2.8
   @nx/angular        : 17.2.8
   @nx/devkit         : 17.2.8
   @nx/eslint-plugin  : 17.2.8
   @nx/node           : 17.2.8
   @nx/plugin         : 17.2.8
   @nx/react          : 17.2.8
   @nx/rollup         : 17.2.8
   @nx/storybook      : 17.2.8
   @nx/vite           : 17.2.8
   @nx/web            : 17.2.8
   @nx/webpack        : 17.2.8
   typescript         : 5.2.2
   ---------------------------------------
   Community plugins:
   @openadp/tools-plugin : 24.5.0
   @storybook/angular    : 7.6.10
   ---------------------------------------
   Local workspace plugins:
   	 @openadp/tools-plugin
```

### Package Manager Version

pnpm  8.14.3

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

This problem exists with any executor that uses temporary tsconfigs. 

The webpack dev server does this correctly, https://github.com/nrwl/nx/blob/master/packages/webpack/src/executors/dev-server/dev-server.impl.ts#L57

Alternatively, there needs to be a way to tell the executor what build target to use for calculating dependencies.

**Comment by leosvelperez** [comment]
Thanks for reporting this!

Could you try running the build setting the `NX_BUILDABLE_LIBRARIES_TASK_GRAPH=true` env var and see if it helps? We are close to switching the behavior it enables to be the default, but we'd like to get some more feedback on existing workspaces that face this issue.

**Comment by leosvelperez** [comment]
hey @statop! 👋🏻 

Have you tried running your builds with the `NX_BUILDABLE_LIBRARIES_TASK_GRAPH=true` env var?

**Comment by statop** [comment]
getting errors 
<img width="1643" alt="Screenshot 2024-02-13 at 11 54 08 AM" src="https://github.com/nrwl/nx/assets/2013403/01274bce-3cf1-4573-b82d-e4f15cee6e1f">

**Comment by leosvelperez** [comment]
I see. It seems you have a project depending on another project that's not a library. That said, we have a bug and are incorrectly handling this scenario. I'm pushing [a fix for that](https://github.com/nrwl/nx/pull/22015). Once it's released, I'll let you know so you can try again with the env var.

**Comment by leosvelperez** [comment]
hey @statop!

The fix was released in Nx 18.0.6. Could you give it another go using that version and running your build with `NX_BUILDABLE_LIBRARIES_TASK_GRAPH=true` set?
